src/http-api.ts

```diff
@@ -128,6 +128,29 @@ export class HttpApi {
       }
     });
 
+    this.#api.get('/:did/query', async (req, res) => {
+      const options = {} as any;
+      for (const param in req.query as object) {
+        const keys = param.split('.');
+        const lastKey = keys.pop();
+        keys.reduce((obj, key) => obj[key] = obj[key] || {}, options)[lastKey] = req.query[param]
+      }
+      const record = await RecordsQuery.create({
+        filter: options.filter,
+        pagination: options.pagination,
+        dateSort: options.dateSort,
+      });
+      const reply = (await this.dwn.processMessage(req.params.did, record.toJSON())) as RecordsQueryReply;
```

Comment:
No need to do ugly type cast:
```suggestion
      const reply = await this.dwn.processMessage(req.params.did, record.message);
```